client/thin-replica-client/include/client/thin-replica-client/grpc_connection.hpp

```diff
@@ -59,35 +62,46 @@ const int kGrpcMaxInboundMsgSizeInBytes = 1 << 24;
 // Server, abstracting the connection and communication implementations from the
 // core Thin Replica Client logic in thin_replica_client::ThinReplicaClient.
 //
-// The TrsConnection itself provides the production implementation used for
-// communication by the Thin Replica Client Library, however, TrsConnection also
+// The GrpcConnection itself provides the production implementation used for
+// communication by the Thin Replica Client Library, however, GrpcConnection also
 // defines the interface (with its virtual functions) that the ThinReplicaClient
 // implementation abstractly uses for communication, so it can be extended in
 // order to use an alternative connection and communication implementation (for
 // example, for testing).
 //
-// TrsConnection provides NO thread safety guarantees in the event more than one
-// call to a TrsConnection function executes concurrently on the same
-// TrsConnection object.
-class TrsConnection {
+// GrpcConnection provides NO thread safety guarantees in the event more than one
+// call to a GrpcConnection function executes concurrently on the same
+// GrpcConnection object.
+class GrpcConnection {
  public:
-  // Possible results of RPC operations a TrsConnection may report.
-  enum class Result { kUnknown, kSuccess, kFailure, kTimeout, kOutOfRange, kNotFound };
-
-  TrsConnection(const std::string& address,
-                const std::string& client_id,
-                uint16_t data_operation_timeout_seconds,
-                uint16_t hash_operation_timeout_seconds)
+  // Possible results of RPC operations a GrpcConnection may report.
+  enum class Result { kUnknown, kSuccess, kFailure, kTimeout, kOutOfRange, kNotFound, kEndOfStream };
+
+  GrpcConnection(const std::string& address,
+                 const std::string& client_id,
+                 uint16_t data_operation_timeout_seconds,
+                 uint16_t hash_operation_timeout_seconds,
+                 uint16_t snapshot_operation_timeout_seconds)
       : logger_(logging::getLogger("concord.client.thin_replica.trscon")),
         address_(address),
         client_id_(client_id),
         data_timeout_(std::chrono::seconds(data_operation_timeout_seconds)),
-        hash_timeout_(std::chrono::seconds(hash_operation_timeout_seconds)) {}
-  virtual ~TrsConnection() { this->disconnect(); }
+        hash_timeout_(std::chrono::seconds(hash_operation_timeout_seconds)),
+        snapshot_timeout_(std::chrono::seconds(snapshot_operation_timeout_seconds)) {}
+  // If the snapshot operation is not needed, then it will not be provided and default of
+  // 5 sec will be taken.
+  GrpcConnection(const std::string& address,
+                 const std::string& client_id,
+                 uint16_t data_operation_timeout_seconds,
+                 uint16_t hash_operation_timeout_seconds)
+      : GrpcConnection(address, client_id, data_operation_timeout_seconds, hash_operation_timeout_seconds, 5) {}
```

Comment:
Why do we need a new constructor? Instead of setting `snapshot_operation_timeout_seconds=5`?

Reply:
thanks, @robem I will do this and keep one ctor.